distributed/worker_state_machine.py

```diff
@@ -2859,7 +2869,9 @@ def _handle_compute_task(self, ev: ComputeTaskEvent) -> RecsInstrs:
                 )
             )
         elif ts.state == "error":
-            instructions.append(TaskErredMsg.from_task(ts, stimulus_id=ev.stimulus_id))
+            instructions.append(
+                TaskErredMsg.from_task(ts, run_id=ev.run_id, stimulus_id=ev.stimulus_id)
+            )
```

Comment:
coverage says we never test this use case.
This is not something you did; but would you mind taking the opportunity to add a unit test for it? It's a fairly big blind spot.

Reply:
I'm not even sure if the code here is correct. I would assume that in this case (especially with the task-erred signing) this task should rather be released and recomputed instead of being reported erred.

Reply:
Agreed, I think that in this case as well as the case above for `ts.state == "memory"`, we should probably recompute instead of acting based on the existing result. I will create a follow-up issue.